Vickabu/social-media-client
# Workflow tools - Add EsLint, Prettier, commit hooks, Jest, Babel, and Cypress setup

**This PR sets up essential tools and configurations for code quality, formatting, and testing, including linting, formatting, commit hooks, and test frameworks.** 

**The main changes include:**

**EsLint and Prettier:**
- Installed and configured EsLint for code quality checks and Prettier for automatic code formatting.
- Added linting and formatting rules according to our coding standards.

**Commit Hooks (Husky):**
- Set up Husky to run linting and formatting on commit (pre-commit) and lint checks on push (pre-push).

**Jest and Babel:**
- Installed Jest for unit testing and Babel to support modern JavaScript features in tests.

**Cypress:**
- Installed Cypress for end-to-end testing.

**How to test:**
- Run npm run lint to verify the linting rules.
- Run npm test to execute the Jest unit tests.
- Run Cypress through the command line (npm run test-e2e) to start the end-to-end tests.
